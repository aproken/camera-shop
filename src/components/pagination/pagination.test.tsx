import { render, RenderResult } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import Pagination from './pagination';

test('Тестирование компонента Pagination', () => {
  const currentPage = 2;
  const pageNumbers = [1, 2, 3, 4, 5];

  const { container }: RenderResult = render(
    <MemoryRouter>
      <Pagination currentPage={currentPage} pageNumbers={pageNumbers} />
    </MemoryRouter>
  );

  expect(container).toMatchSnapshot();
});
