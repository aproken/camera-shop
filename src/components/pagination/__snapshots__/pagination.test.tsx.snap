// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Тестирование компонента Pagination 1`] = `
<div>
  <div
    class="pagination"
  >
    <ul
      class="pagination__list"
    >
      <li
        class="pagination__item"
      >
        <a
          class="pagination__link pagination__link--text"
          href="/page/1"
        >
          Назад
        </a>
      </li>
      <li
        class="pagination__item"
      >
        <a
          class="pagination__link"
          href="/page/1"
        >
          1
        </a>
      </li>
      <li
        class="pagination__item"
      >
        <a
          class="pagination__link pagination__link--active"
          href="/page/2"
        >
          2
        </a>
      </li>
      <li
        class="pagination__item"
      >
        <a
          class="pagination__link"
          href="/page/3"
        >
          3
        </a>
      </li>
      <li
        class="pagination__item"
      >
        <a
          class="pagination__link"
          href="/page/4"
        >
          4
        </a>
      </li>
      <li
        class="pagination__item"
      >
        <a
          class="pagination__link"
          href="/page/5"
        >
          5
        </a>
      </li>
      <li
        class="pagination__item"
      >
        <a
          class="pagination__link pagination__link--text"
          href="/page/3"
        >
          Далее
        </a>
      </li>
    </ul>
  </div>
</div>
`;
